crates/core/src/transcript.rs

```diff
@@ -631,4 +631,207 @@ mod tests {
         let subseq = transcript.get(Direction::Sent, &Idx(RangeSet::from([0..4, 7..10, 11..13])));
         assert_eq!(subseq, None);
     }
+
+    #[rstest]
+    fn test_transcript_to_partial_success(transcript: Transcript) {
+        let partial = transcript.to_partial(Idx::new(0..2), Idx::new(3..7));
+        assert_eq!(partial.sent_unsafe(), [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]);
+        assert_eq!(
+            partial.received_unsafe(),
+            [0, 0, 0, 3, 4, 5, 6, 0, 0, 0, 0, 0]
+        );
+    }
+
+    #[rstest]
+    #[should_panic]
+    fn test_transcript_to_partial_failure(transcript: Transcript) {
+        let _ = transcript.to_partial(Idx::new(0..14), Idx::new(3..7));
+    }
+
+    #[rstest]
+    fn test_partial_transcript_contains(transcript: Transcript) {
+        let partial = transcript.to_partial(Idx::new(0..2), Idx::new(3..7));
+        assert!(partial.contains(Direction::Sent, &Idx::new([0..5, 7..10])));
+        assert!(!partial.contains(Direction::Received, &Idx::new([4..6, 7..13])))
+    }
+
+    #[rstest]
+    fn test_partial_transcript_unauthed(transcript: Transcript) {
+        let partial = transcript.to_partial(Idx::new(0..2), Idx::new(3..7));
+        assert_eq!(partial.sent_unauthed(), Idx::new(2..12));
+        assert_eq!(partial.received_unauthed(), Idx::new([0..3, 7..12]));
+    }
+
+    #[rstest]
+    fn test_partial_transcript_union_success(transcript: Transcript) {
+        // non overlapping ranges
```

Comment:
@yuroitaki , could you pls make sure all comments starts with a Capital letter and have a period at the end. This formating rule applies not just to /// comments but also to // comments like these.
as per https://github.com/tlsnotary/tlsn/blob/main/CONTRIBUTING.md#capitalization-and-punctuation